### PR TITLE
Change plugin templates

### PR DESCRIPTION
**Pull request by sevisal**
Modify plugin templates to instantiate modules within the plugin's main script

### CODE REVIEW

File: src/vai_lab/DataProcessing/DataProcessing_core.py
Comment (teemu-ruokolainen) [inline src/vai_lab/DataProcessing/DataProcessing_core.py:1]: At line 14: 

Does `data_in` have a type? Could it be given as a type hint?

`def _load_plugin(self, data_in) -> None:`

The old line:

`def _load_plugin(self, plugin_name: str) -> None:`

File: src/vai_lab/Environment/Environment_core.py [hunk 1]
```diff
@@ -11,31 +11,29 @@ def set_avail_plugins(self, avail_plugins: PluginSpecsInterface) -> None:
     def set_data_in(self, data_in: DataInterface) -> None:
         self._data_in = data_in
 
-    def _load_plugin(self, plugin_name: str) -> None:
+    def _load_plugin(self, data_in) -> None:
+        self._plugin_name = self._module_config["plugin"]["plugin_name"]
```
Comment (teemu-ruokolainen) [inline src/vai_lab/Environment/Environment_core.py:15]: Is this `self._plugin_name` variable assigned here used often? Is it a convenience variable? It looks like the same information can always be accessed directly from `self._module_config["plugin"]["plugin_name"]?

File: src/vai_lab/DataProcessing/DataProcessing_core.py
Comment (sevisal) [reply src/vai_lab/DataProcessing/DataProcessing_core.py:1]: It's a `DataInterface` type. I have just added it

File: src/vai_lab/Environment/Environment_core.py (hunk 1 above)
Comment (sevisal) [reply src/vai_lab/Environment/Environment_core.py]: It was thought to be a convenience variable, but you are right we never used it. I just removed it.

**Review by teemu-ruokolainen — APPROVED** [review]
Just a few mostly aesthetic comments, otherwise looks ok.

**Pull request merged**